src/test/server/store-test.js

```diff
@@ -1,9 +1,203 @@
+    });
+}
+
+function _getInitialUserList () {
+    return usersData.map((user) => {
+        return new UserInfo(
+            {
+                id:      user.id,
+                name:    user.name,
+                phone:   user.phone,
+                email:   user.email,
+                website: user.website
+            },
+            user.address,
+            user.company
+        );
+    });
+}
 
 describe('UserStore', () => {
+    afterEach(() => {
+        UserStore._users        = _getInitialUserList();
+        UserStore._searchedInfo = null;
+        UserStore.removeAllListeners(EVENT_TYPE.userAdded);
+        UserStore.removeAllListeners(EVENT_TYPE.addingFailed);
+        UserStore.removeAllListeners(EVENT_TYPE.change);
+        UserStore.removeAllListeners(EVENT_TYPE.usersFound);
+    });
+
     describe('.getUsers()', () => {
         it('Should return a list of users', () => {
-            console.log(UserStore.getUsers());
+            const users = _getInitialUserList();
+
+            assert.deepEqual(UserStore.getUsers(), users);
+        });
+    });
+
+    describe('Actions', () => {
+        describe('addNewUser', () => {
+            it('Should emit userAdded event', async () => {
+                let newUser = TEST_USER;
+
+                newUser.id = UserStore.getUsers().length + 1;
+                await subEvent(EVENT_TYPE.userAdded, () => UserAction.addNewUser(TEST_USER));
+
+                assert.deepEqual(UserStore._users[UserStore._users.length - 1], newUser);
+            });
+
+            it('Should emit addingFailed event with duplicateUserId if user exists ', async () => {
+                const prevUserStoreSize = UserStore.getUsers().length;
+                const [userId]          = await subEvent(EVENT_TYPE.addingFailed, () => UserAction.addNewUser(TEST_USER_EXISTING));
+
+                assert(UserStore.getUsers().length, prevUserStoreSize);
+                assert.strictEqual(userId, 2);
+            });
+
+            it('Should emit userAdded event if user exists, but force flag is true', async () => {
+                let newUser = TEST_USER_EXISTING;
+
+                newUser.id = UserStore.getUsers().length + 1;
+                await subEvent(EVENT_TYPE.userAdded, () => UserAction.addNewUser(TEST_USER_EXISTING, true));
+
+                assert.deepEqual(UserStore._users[UserStore._users.length - 1], newUser);
+            });
+
+            it('Should emit usersFound event with updated foundUsers if we are in search mode', async () => {
+                const prevUserStoreSize = UserStore.getUsers().length;
+
+                let newUser = TEST_USER;
+
+                newUser.id                  = prevUserStoreSize + 1;
+                UserStore._searchedInfo     = SEARCH_USER_INFO;
+                const [foundUsers, ...args] = await subEvent(EVENT_TYPE.usersFound, () => UserAction.addNewUser(TEST_USER));
+
+                assert(foundUsers.includes(TEST_USER));
```

Comment:
the more strict is our assertion the better
`assert.deepEqual(foundUsers, [TEST_USER]); 
`assert.deepEqual(UserStore._users[UserStore._users.length - 1], newUser);`

Reply:
Don't you want to check highlighted fields?
the same for `'Should emit usersFound event with updated foundUsers and searchFields if update in search mode`